sources/packages/backend/apps/api/src/route-controllers/application/application.students.controller.ts

```diff
@@ -202,6 +205,17 @@ export class ApplicationStudentsController extends BaseController {
     ) {
       throw new UnprocessableEntityException("Invalid offering intensity.");
     }
+    if (payload.data.selectedProgram) {
```

Comment:
Please look into the complexity reported by sonar cloud. 

![image](https://github.com/bcgov/SIMS/assets/54600590/5312e132-9485-481a-8bd3-5cd9f494dc59)